lib/capistrano/hivequeen/deploy.rb

```diff
@@ -7,13 +7,23 @@
   before "deploy:stage", "hivequeen:start"
   before 'hivequeen:start', 'hivequeen:check_commit'
   on :start, "hivequeen:require_environment", :except => HiveQueen.environment_names
+  on :start, "hivequeen:ensure_canary_specifies_hosts"
+
   namespace :hivequeen do
 
     desc "[internal] abort if no environment specified"
     task :require_environment do
       abort "No environment specified." if !exists?(:environment)
     end
 
+    desc "[internal] abort if we're trying to do a canary deploy but HOSTS hasn't been defined"
+    task :ensure_canary_specifies_hosts do
+      # TODO: I suppose we could randomly select instance(s) in this case
+      if canary && !ENV.key?('HOSTS')
+        abort "You asked to do a canary deployment but didn't specify any hosts! \nPlease invoke like `cap HOSTS=foo.com deploy -s canary=true'"
```

Comment:
I like the abort. I feel like it'll at least help in communication of which instance is 'hot', as it were.